### DISCUSSION TITLE
`strapi.config.get('server');` from a plugin component class throws `TypeError: Cannot read property 'get' of undefined`

### DISCUSSION THREAD

**Discussion by WhereJuly** [strapi/strapi#6585] [Legacy Topics]
**Describe the bug**
I try to access the global config values from a plugin React component class `render()` function. It throws `TypeError: Cannot read property 'get' of undefined` exception.

At the same time `strapi` global variable does not bear `config` key neither on itself not on its prototype (which is a generic JS prototype). As well `process.env` is just an empty object (`{}`) if `console.dir(process.env);` it at the same `render()` function.

**Steps to reproduce the behavior**

1. Create a plugin with `strapi generate:plugin my-plugin`;
2. Put console.dir(process.env) and `strapi.config.get('server');` into `my-project/plugins/my-plugin/admin/src/containers/HomePage/index.js` within `HomePage` function;
3. Re-build the admin `npm run build`
4. Open Chrome Dev Tools console and see `process.env` is `{}` and `strapi.config.get('server');` throws `TypeError: Cannot read property 'get' of undefined` exception.

**Expected behavior**
1. As implied in the [Configuration docs](https://strapi.io/documentation/v3.x/concepts/configurations.html#configuration) the `strapi` object should contain `config` key with `get()` method on it.

**System**
- Node.js version: 12.16.1
- NPM version: 6.14.4
- Strapi version: 3.0.1
- Database: PostgreSQL 10
- Operating system: Windows 8.1

**Comment by derrickmehaffy** [comment]
Moving this over to a discussion as it's not a bug with the Strapi source code but a custom plugin.

**Reply by WhereJuly** [reply]
@derrickmehaffy Are you sure the fact that the global `strapi` config is not available in Strapi plugins is not a bug? I doubt seriously.

**Reply by WhereJuly** [reply]
@derrickmehaffy Then how am I supposed to access my global Strapi configuration from plugin? This is definitely a bug.

**Reply by derrickmehaffy** [reply]
Are you trying to access it from the "frontend" of the plugin?

If so that is not supported, and none of the existing plugins do this. You will need to write controllers to pull the data you want from the plugin's "backend"

**Reply by derrickmehaffy** [reply]
https://strapi.io/documentation/v3.x/plugin-development/plugin-architecture.html#plugin-folders-and-files-architecture

**Reply by derrickmehaffy** [reply]
These are the only supported paths that the "Frontend" of the plugin can access (which are populated during the build process): https://strapi.io/documentation/v3.x/plugin-development/frontend-development.html#api

**Reply by WhereJuly** [reply]
@derrickmehaffy Yes. I am trying to acces from the plugin's frontend. I see your point. Thanks for clarification.
